asserts/repair.go

```diff
@@ -131,6 +153,38 @@ func assembleRepair(assert assertionBase) (Assertion, error) {
 	if err != nil {
 		return nil, err
 	}
+	modes, err := checkStringList(assert.headers, "modes")
+	if err != nil {
+		return nil, err
+	}
+	bases, err := checkStringList(assert.headers, "bases")
+	if err != nil {
+		return nil, err
+	}
+
+	// verify that modes is a list of only "run", "recover" and "install"
+	if len(modes) != 0 {
+		for _, m := range modes {
+			// note we could import boot here to use i.e. boot.ModeRun, but that
+			// might make import cycles happen some day if boot needs to use
+			// the asserts package for some odd reason, so instead just use the
+			// values directly, they're unlikely to change now
+			if !strutil.ListContains([]string{"run", "recover", "install"}, m) {
```

Comment:
There's always tension between letting people create nonsensical assertions vs preventing a old snapd to interpret a corner case for which we found a use case later but in this case whatever we do here needs to be supported n the repair services and runner. So I think we should match what we plan to do for now allow only "run" and "recover" here.

Reply:
done, now install is invalid and we unit test this